cluster-autoscaler/core/scale_down.go

```diff
@@ -697,6 +720,7 @@ func (sd *ScaleDown) TryToScaleDown(allNodes []*apiv1.Node, pods []*apiv1.Pod, p
 		// Finishing the delete process once this goroutine is over.
 		defer sd.nodeDeleteStatus.SetDeleteInProgress(false)
 		err := sd.deleteNode(toRemove.Node, toRemove.PodsToReschedule)
+		sd.nodeDeleteStatus.AddNodeDeleteError(toRemove.Node.Name, err)
```

Comment:
Do we still want to do this if err == nil?

Reply:
Can we name it sth like result instead, then?

Also, is there any reason not to defer this call? That way, waiting for lock won't stop logging error and updating metrics.